src/main.rs

```diff
@@ -36,34 +36,55 @@ fn main() -> Result<()> {
     // Create a stream for capturing the output
     let output_stream = wasi_common::pipe::WritePipe::new_in_memory();
 
+    // Create a stream for capturing the error logs
+    let error_stream = wasi_common::pipe::WritePipe::new_in_memory();
+
     {
         // Link WASI and construct the store.
         let mut linker = Linker::new(&engine);
         wasmtime_wasi::add_to_linker(&mut linker, |s| s)?;
         let wasi = WasiCtxBuilder::new()
             .stdin(Box::new(input_stream))
             .stdout(Box::new(output_stream.clone()))
+            .stderr(Box::new(error_stream.clone()))
             .inherit_args()?
             .build();
         let mut store = Store::new(&engine, wasi);
 
         linker.module(&mut store, "", &module)?;
 
         // Execute the module
-        linker
+        let result = linker
             .get_default(&mut store, "")?
             .typed::<(), (), _>(&store)?
-            .call(&mut store, ())?;
+            .call(&mut store, ());
```

Comment:
We could probably just panic here and exit if this call to the main script fn fails I think. Otherwise, I can see the decoding on line 86 failing cause we wouldn't have a proper output

Reply:
👍🏼 if you found it useful then makes sense to keep